Libraries/Components/Touchable/TouchableOpacity.js

```diff
@@ -67,16 +67,24 @@ var TouchableOpacity = React.createClass({
   },
 
   getInitialState: function() {
+    var childStyle = flattenStyle(this.props.style) || {};
+    var opacity = childStyle.opacity === undefined ? 1 : childStyle.opacity;
     return {
       ...this.touchableGetInitialState(),
-      anim: new Animated.Value(1),
+      anim: new Animated.Value(opacity),
     };
   },
 
   componentDidMount: function() {
     ensurePositiveDelayProps(this.props);
   },
 
+  componentDidUpdate: function() {
+    if (!this._hideTimeout && !this._pressIn) {
+      this._opacityInactive();
+    }
+  },
```

Comment:
Can you add a test to ensure that this continues to work after you get your fix in?